src/test/testrunner.ts

```diff
@@ -364,10 +364,9 @@ class JDCommandEvaluator {
                 )
                 const ev = expr.eval(args[1])
                 const reg = args[0] as jsep.Identifier
-                if (this.testRunner.serviceTestRunner.writeRegister(reg.name, ev)) {
-                    this._status = JDTestCommandStatus.Passed
-                    this._progress = `wrote ${ev} to register ${reg.name}`
-                }
+                await this.testRunner.serviceTestRunner.writeRegisterAsync(reg.name, ev)
+                this._status = JDTestCommandStatus.Passed
```

Comment:
you should use a property setter and automatically emit a change event when status change
same for progress.